os/include/sys/types.h

```diff
@@ -187,10 +187,14 @@ typedef int16_t ssize_t;
 typedef uint16_t rsize_t;
 
 #else							/* CONFIG_SMALL_MEMORY */
+/* As a general rule, the size of size_t should be the same as the size of
+ * uintptr_t: 32-bits on a machine with 32-bit addressing but 64-bits on a
+ * machine with 64-bit addressing.
+ */
 
-typedef uint32_t size_t;
-typedef int32_t ssize_t;
-typedef uint32_t rsize_t;
+typedef uintptr_t size_t;
```

Comment:
uintptr_t, intptr_t  & part of C standard which defines the role for this and not apt above.